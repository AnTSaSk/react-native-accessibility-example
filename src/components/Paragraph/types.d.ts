import { ReactNode } from 'react';

import { themeStyleProps } from '../../hooks/types.d';

export interface ParagraphProps {
  className?: string;
  isDark?: boolean;
  theme?: themeStyleProps;
  children: ReactNode;
}
