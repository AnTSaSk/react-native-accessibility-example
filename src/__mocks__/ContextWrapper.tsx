import React, { FunctionComponent } from 'react';
import { AppearanceProvider } from 'react-native-appearance/src/mock';

// Context
import { AccessibilityProvider } from '../hooks/Accessibility';
import { ThemeProvider } from '../hooks/Theme';

const ContextWrapper: FunctionComponent<{}> = ({ children }) => (
  <AppearanceProvider>
    <AccessibilityProvider>
      <ThemeProvider>{children}</ThemeProvider>
    </AccessibilityProvider>
  </AppearanceProvider>
);

export default ContextWrapper;
