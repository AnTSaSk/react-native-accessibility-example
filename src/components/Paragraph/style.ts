import styled from '@emotion/native';

export const ParagraphComponent = styled.Text`
  margin: 5px 0;
  color: ${(props) =>
    props.isDark ? props.theme.colors.white : props.theme.colors.black};
  font-size: 16px;
`;
