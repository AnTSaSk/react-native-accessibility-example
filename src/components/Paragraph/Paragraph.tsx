import React, { StatelessComponent } from 'react';

// Context
import { useTheme } from '../../hooks/theme';

// Types
import { ParagraphProps } from './types';

// Styles
import { ParagraphComponent } from './style';

const Paragraph: StatelessComponent<ParagraphProps> = (props) => {
  const { theme } = useTheme();

  return (
    <ParagraphComponent
      isDark={theme === 'dark'}
      accessible
      accessibilityRole="text"
      {...props}
    >
      {props.children}
    </ParagraphComponent>
  );
};

export default Paragraph;
