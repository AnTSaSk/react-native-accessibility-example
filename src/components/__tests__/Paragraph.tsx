import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

// Mocked context
import ContextWrapper from '../../__mocks__/ContextWrapper';

// Component
import Paragraph from '../Paragraph';

const TestComponent = (
  <ContextWrapper>
    <Paragraph>Test content</Paragraph>
  </ContextWrapper>
);

it('Renders correctly Paragraph component', () => {
  const paragraph = renderer.create(TestComponent).toJSON();

  expect(paragraph).toMatchSnapshot();
});
