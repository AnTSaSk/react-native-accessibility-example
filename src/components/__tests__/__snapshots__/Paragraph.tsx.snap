// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly Paragraph component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityRole="text"
    accessible={true}
    isDark={false}
    style={
      Object {
        "color": "#000000",
        "fontSize": 16,
        "marginBottom": 5,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 5,
      }
    }
  >
    Test content
  </Text>
</View>
`;
